xDripG5/Transmitter.swift

```diff
@@ -30,7 +30,12 @@ public final class Transmitter: BluetoothManagerDelegate {
 
     /// The ID of the transmitter to connect to
     public var ID: String {
-        return id.id
+        get {
+            return id.id
+        }
+        set(newID) {
```

Comment:
Adding a public setter to Transmitter.id isn’t the right solution, since additional state on Transmitter like activationDate need to be reset. I think it’s cleaner for clients to recreate Transmitter when the ID changes.

Reply:
Thanks @ps2, will do, much cleaner. I have changed the client code to recreate Transmitter; it just doesn't always start scanning reliably after a change to the transmitter ID. Once I've chased that down I'll push the changes.

Reply:
What if we kept the setter, but put all state in the TransmitterID object (and rename it to something sensible). AFAIK we just need to associate `ID`, `activationDate` and `lastTimeMessage`with a particular transmitter. I can't work out how to clean up all CBCentralManager state if we switch out the whole Transmitter object.

Reply:
What if we store the transmitter state (`activationDate` and `lastTimeMessage` in the TransmitterID object, which gets renewed on a new ID). If we write the setter like this
```
        set(newID) {
            stopScanning()
            id = TransmitterID(id: newID)
            resumeScanning()
        }
```
then we scan afresh when the user changes ID.

I'd be happy to replace the entire Transmitter object, but can't find a clean way for the old Transmitter object to release all bluetooth resources in time for the new object to use them (if that makes sense).

Reply:
We do need to reset the bluetooth peripheral if we're changing the transmitter id.

Reply:
I have a set of patches that incorporate what you're trying to do here... will get them posted soon.

Reply:
See #70; I've incorporated some of the things you've been pointing out there.  I think this PR should just focus on getting the example app building.  I.e. Removing cocoapods, and incorporating the example app into the project.

Reply:
👍 thanks!